avocado/utils/cpu.py

```diff
@@ -129,6 +129,8 @@ def get_cpu_arch():
     for (pattern, arch) in cpu_table:
         if _list_matches(cpuinfo, pattern):
             return arch
+    if "CPU architecture" in cpuinfo:
+        return platform.machine()
```

Comment:
Well, this changes the returned value from `arm` to `aarch64`. I guess it's actually intended, but how about simply using `platform.machine()` as default and only proceed with custom logic in case `ppc` is detected? That should give actually even better results.

Yes, the problem of this function is it's abused to return CPU type on ppc and architecture on other architectures, but we have to cope with that...

Reply:
@ldoktor For 32 bit arm, as ARM pattern will be found in the cpuinfo (ARMv7, etc.) it will return arm. So I did not change the logic for 32 bit.

On 32bit arm, with testing on one raspberry pi, platfrom.machine return armv7l, not arm..., that means other code with equal arm will fail...

For 64 bit, no match of the pattern in cpuinfo, then in match with "CPU architecture" it will return platform.machine which for GNU/linux will be aarch64.

Yeah, the ppc part is another cause of not directly replace with platform.machine.

Reply:
actually the `armv7l` return would be closer to `ppc` returns of this function. I'm really tempted to either obsolete this function as it provides inconsistent results and add 2 functions one `platform.machine` and another one with just the `arm|aarch64|ppc64le|s390x|...`. Anyway I'll bring this up on our meeting and as we need to support this function in the mean-time and your fix works well I'm merging it...